### ISSUE TITLE
Store HTTP headers in history

### ISSUE THREAD

**Issue by pyrog** [OpenRefine/OpenRefine#3234]
When reusing an expression in the history for **adding a column by fetching URLs**, one must re-enter `Authorization`, `User-Agent`and `Accept` parameters.

### Proposed solution
Store HTTP parameters in the History / Starred.

### Alternatives considered
Store them in the settings of OpenRefine and display a list drop down list to choose them

Related to #218

**Comment by tfmorris** [comment]
`Authorization` headers should never be stored in the history (or logged anywhere else). That's a big security risk.

I actually thought I'd seen HTTP headers in the operations history recently, so I'll double check on the behavior.

**Comment by pyrog** [comment]
Not logged anywhere, yes. But could be encrypted somewhere in the project.

The project itself could be encrypted because it could contain sensible data 😉

**Comment by thadguidry** [comment]
@tfmorris Do you think maybe the proposed solution could be altered to allow storing them as encrypted variables?
Many CI/CD platforms and Automation frameworks use variables for this purpose for reuse.  They ensure that those variables that are credential related are stored encrypted and that those credential related variables are never output into the logging system or at least masked. (my experience is that of [Gitlab variables](https://docs.gitlab.com/ee/ci/variables/#create-a-custom-variable-in-the-ui) which also have masking)   We could do the same somehow?

**Comment by ostephens** [comment]
> Authorization headers should never be stored in the history (or logged anywhere else). That's a big security risk.
> I actually thought I'd seen HTTP headers in the operations history recently, so I'll double check on the behavior.

You are correct - that's my fault I'm afraid. Logging the headers was deliberate, but we need a way of handling the authorisation that doesn't expose credentials. Possibly the user needs prompting for Authorisation when the request requires it?

**Comment by wetneb** [comment]
@ostephens you did the right thing given the infrastructure we have in place at the moment - to get it right, we would need a better mechanism to supply secret data to operations.
